drive_test.go

```diff
@@ -0,0 +1,361 @@
+package deta
+
+import (
+	"bytes"
+	"crypto/rand"
+	"errors"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+const (
+	readChunkSize = 1024 * 1024 * 10
+)
+
+func SetupDrive() *Drive {
+	projectKey := os.Getenv("DETA_SDK_TEST_PROJECT_KEY")
+	driveName := os.Getenv("DETA_SDK_TEST_DRIVE_NAME")
+	rootEndpoint := os.Getenv("DETA_SDK_TEST_ENDPOINT")
+	return newDrive(projectKey, driveName, rootEndpoint)
+}
+
+func TearDownDrive(d *Drive, t *testing.T) {
+	lr, err := d.List(1000, "", "")
+	if !errors.Is(err, nil) {
+		t.Log("Failed to list names in teardown, further tests might fail")
+	}
+	for _, name := range lr.Names {
+		_, err = d.Delete(name)
+		if !errors.Is(err, nil) {
+			t.Logf("Failed to delete test file with name '%s'.\nFurther tests might fail", name)
+		}
+	}
+
+}
+
+func TestPutDrive(t *testing.T) {
+	drive := SetupDrive()
+	defer TearDownDrive(drive, t)
+
+	testCases := []struct {
```

Comment:
can you please also put test cases with
1. empty content type
2. other io.Readers besides `strings.NewReader`